Sources/Features/Breadcrumb/QueueFile.swift

```diff
@@ -0,0 +1,45 @@
+import Foundation
+
+@objcMembers
+public class Queue<T>: NSObject {
+    private var elements: [T] = []
+
+    func enqueue(_ element: T) {
+        elements.append(element)
+    }
+
+    func dequeue() -> T? {
+        if elements.isEmpty {
+            return nil
+        } else {
+            return elements.removeFirst()
+        }
+    }
+
+    func peek() -> T? {
+        return elements.first
+    }
+
+    func pop() -> T? {
```

Comment:
are we sure in the queue we should have a pop method at all?

Reply:
just a remove last will do I think, you have another suggestion?